libs/ledger-live-common/src/families/tezos/bakers.ts

```diff
@@ -132,28 +159,32 @@ export function getAccountDelegationSync(account: AccountLike): Delegation | nul
     receiveShouldWarnDelegation,
   };
 }
+
 export function isAccountDelegating(account: AccountLike): boolean {
   return !!getAccountDelegationSync(account);
 }
-export async function loadBaker(addr: string): Promise<Baker | null | undefined> {
+
+export async function loadBaker(addr: string): Promise<Baker | undefined> {
   const cacheBaker = getBakerSync(addr);
-  if (cacheBaker) return Promise.resolve(cacheBaker);
+  if (cacheBaker) return cacheBaker;
```

Comment:
In that case we don't return a promise right ? so the return type is Promise | Baker ?

Reply:
Actually if you have an `async` function whatever you return inside this function is a Promise